src/mds/Server.cc

```diff
@@ -1046,6 +1102,93 @@ void Server::respond_to_request(MDRequestRef& mdr, int r)
     mdr->internal_op_finish->complete(r);
     mdcache->request_finish(mdr);
   }
+
+  // add here to avoid counting ops multiple times (e.g., locks, loading)
+  switch(mdr->client_request->get_op()) {
```

Comment:
I think all the ops you've got stats for are client ones (that includes snapshots/layout stuff).  The internal ones are listed in include/ceph_fs.h line 363 onwards.